ext/date/php_date.c

```diff
@@ -3968,24 +3968,22 @@ PHP_FUNCTION(date_diff)
 }
 /* }}} */
 
-static bool timezone_initialize(php_timezone_obj *tzobj, const char *tz, size_t tz_len, char **warning_message) /* {{{ */
+static bool timezone_initialize(php_timezone_obj *tzobj, const zend_string *tz_zstr, char **warning_message) /* {{{ */
 {
 	timelib_time *dummy_t = ecalloc(1, sizeof(timelib_time));
 	int           dst, not_found;
-	const char   *orig_tz = tz;
+	const char *tz = ZSTR_VAL(tz_zstr);
 
-	if (strlen(tz) != tz_len) {
-		if (warning_message) {
-			spprintf(warning_message, 0, "Timezone must not contain null bytes");
-		}
+	if (UNEXPECTED(zend_str_has_nul_byte(tz_zstr))) {
```

Comment:
The cleaner, and faster, approach is to move this check simply to php_date_timezone_initialize_from_hash IMO.